src/models/highlighter.cpp

```diff
@@ -19,13 +19,16 @@
 #include <KSyntaxHighlighting/Theme>
 #endif
 
-Highlighter::Highlighter(QTextDocument* document, QObject* parent)
+Highlighter::Highlighter(QTextDocument* document, KSyntaxHighlighting::Repository* repository, QObject* parent)
     : QObject(parent)
 #if KF5SyntaxHighlighting_FOUND
-    , m_repository(std::make_unique<KSyntaxHighlighting::Repository>())
     , m_highlighter(new KSyntaxHighlighting::SyntaxHighlighter(document))
+    , m_repository(repository)
 #endif
 {
+#if KF5SyntaxHighlighting_FOUND == 0
+    Q_UNUSED(repository);
+#endif
```

Comment:
```
#else
    Q_ASSERT(repository);
```

Reply:
otherwise the code below would crash where `m_repository` is accessed without checks

I think this might actually lead to crashes in the unit tests when `updateColorTheme` is called below?